adserver/analyzer/tasks.py

```diff
@@ -63,46 +67,59 @@ def analyze_url(url, publisher_slug, force=False):
             response = backend_instance.fetch()
 
         analyzed_keywords = backend_instance.analyze(response)  # Can be None
-        log.debug("Keywords from '%s': %s", backend.__name__, analyzed_keywords)
+        if analyzed_keywords:
+            log.debug("Keywords from '%s': %s", backend.__name__, analyzed_keywords)
+
+            for kw in analyzed_keywords:
+                keywords.add(kw)
 
         analyzed_embedding = backend_instance.embedding(response)  # Can be None
```

Comment:
I do wonder if it makes sense to break embedding out into it's own workflow. It's kinda grown beyond analyzers, and might make more sense outside of this flow? Not 100% sure though, since I do think we probably want to minimize the number of times we download the content.

Reply:
Ya, I've considered this as well. I think it probably could be its own workflow. It'd be nice to have it more self-contained and it could make things more simple for anybody consuming our code but not interested in the analyzers.